analysis/topology/sequences.lean

```diff
@@ -0,0 +1,282 @@
+/-
+Copyright (c) 2018 Jan-David Salchow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jan-David Salchow
+-/
+import analysis.topology.topological_space
+import analysis.metric_space
+import analysis.topology.uniform_space
+import data.set.countable
+import data.real.cau_seq_filter
+
+open filter
+
+/- Sequences in topological spaces.
+ - 
+ - In this file we define sequences in topological spaces and show how they are related to
+ - filters and the topology. In particular, we
+ -
+ - (*) associate a filter with a sequence and show prove equivalence of convergence of the two,
+ - (*) define the sequential closure of a set and prove that it's contained in the closure, 
+ - (*) define a type class "sequential_space" in which closure and sequential closure agree, 
+ - (*) define sequential continuity and show that it coincides with continuity in sequential spaces,
+ - (*) provide and instance that shows that every metric space is a sequential space.
+ -
+ - TODO: There should be an instance that associates a sequential space with a first countable
+ -       space. 
+ - TODO: Sequential compactness should be handeled here.
+ -/
+namespace sequence
+
+universes u v
+variables {X : Type u} {Y : Type v}
+
+def to_filter (x : ℕ → X) : filter X := filter.map x at_top
+
+/- Statements about sequences in general topological spaces. -/
+section topological_space
+variables [topological_space X] [topological_space Y]
+
+/- The notion of convergence of sequences in topological spaces. -/
+def converges_to (x : ℕ → X) (limit : X) : Prop :=
+  ∀ U : set X, limit ∈ U → is_open U → ∃ n0 : ℕ, ∀ n ≥ n0, (x n) ∈ U
+
+lemma const_seq_conv (p : X) : converges_to (λ n, p) p :=
+assume U (_: p ∈ U) (_ : is_open U), exists.intro 0 (assume n (_ : n ≥ 0), ‹p ∈ U›)
```

Comment:
When you don't use an assumption like `is_open U`, it's sometimes cleaner to use only a placeholder: 
`assume U (_ : p ∈ U) _, ...`
Also, space between _ and :

Reply:
Are you sure? This way it's actually readable, not just a weird string...

Reply:
Well, it's irrelevant to the proof either way. I just read `_` as "info that doesn't matter." But it's up to you.

Reply:
For this same reason I find ` (_: p ∈ U) ` confusing, because you *do* use it later, implicitly.